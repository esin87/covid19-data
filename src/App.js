import React from 'react';
import './App.css';
import SearchForm from './components/SearchForm/SearchForm';
import SearchResults from './components/SearchResults/SearchResults';
import Header from './components/Header/Header';
import data from './data.json';
const App = () => {
	return (
		<div className='App'>
			<Header />
			<SearchForm />
			<SearchResults data={data} />
		</div>
	);
};

export default App;
