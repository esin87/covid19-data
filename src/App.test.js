import React from 'react';
import App from './App';
import { mount, render } from 'enzyme';

describe('App component', () => {
	let component;
	beforeEach(() => {
		component = mount(<App />);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('contains a main element with the appropriate text', () => {
		expect(component.find('main').text()).toBe(
			'COVID-19 DataA free, open-source tool to search for up-to-date COVID-19 case information by country. Data courtesy of covid19api.com.Start searching'
		);
	});

	it('contains a footer element with the appropriate text', () => {
		expect(component.find('footer').text()).toBe(
			'Created by Esin Saribudak. See source code here.'
		);
	});
});
