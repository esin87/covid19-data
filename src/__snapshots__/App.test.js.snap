// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match the snapshot 1`] = `"<div class=\\"App\\"><nav style=\\"margin-bottom: 2.5rem;\\" class=\\"navbar navbar-expand-md navbar-light\\"><a href=\\"#/home\\" class=\\"active navbar-brand\\">COVID-19 Data</a><button aria-controls=\\"basic-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"navbar-collapse collapse\\" id=\\"basic-navbar-nav\\"><div class=\\"navbar-nav\\"><a href=\\"#/home\\" data-rb-event-key=\\"#/home\\" class=\\"active nav-link\\">Home</a><a href=\\"#/countries\\" data-rb-event-key=\\"#/countries\\" class=\\"nav-link\\">Country List</a><a href=\\"#/search\\" data-rb-event-key=\\"#/search\\" class=\\"nav-link\\">Country Search</a><a href=\\"#/global\\" data-rb-event-key=\\"#/global\\" class=\\"nav-link\\">Global Summary</a><a href=\\"#/about\\" data-rb-event-key=\\"#/about\\" class=\\"nav-link\\">About</a></div></div></nav><main><div class=\\"container\\"><div class=\\"jumbotron\\"><h1>COVID-19 Data</h1><p>A free, open-source tool to search for up-to-date COVID-19 case information by country. </p><p>Data courtesy of <a href=\\"https://covid19api.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">covid19api.com</a>.</p><p><a href=\\"#/countries\\" class=\\"btn btn-primary\\">Start searching</a></p></div></div></main><footer class=\\"text-muted\\">Created by <a href=\\"https://esinsaribudak.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Esin Saribudak</a>. See source code <a href=\\"https://github.com/esin87/covid19-data\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">here</a>.</footer></div>"`;
